tests/functional/sources/test_source_freshness.py

```diff
@@ -428,11 +436,17 @@ def project_config_update(self):
             },
         }
 
+    @pytest.fixture(scope="class")
+    def global_deprecations(self):
+        deprecations.reset_deprecations()
+        yield
+        deprecations.reset_deprecations()
```

Comment:
this clean-up call was missing from `test_hooks_do_run_for_source_freshness` initially, meaning the global `active_deprecations` was preserved across tests